### PR TITLE
bump golangci-lint / zeitgeist / cosign and dependencies

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

 - bump golangci-lint / zeitgeist / cosign
 - update dependencies

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
 - bump golangci-lint / zeitgeist / cosign
 - update dependencies

```

### CODE REVIEW

File: dependencies.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 dependencies:
   # golangci/golangci-lint
   - name: "golangci-lint"
-    version: 1.57.1
+    version: 1.58.0
```
Comment (saschagrunert) [inline dependencies.yaml:4]: Can we update the config as well?

Comment (cpanato) [reply dependencies.yaml:4]: done, there is anything else in the config that you find we need to update?

Comment (saschagrunert) [reply dependencies.yaml:4]: Yes, you can apply this patch:

```patch
diff --git a/.golangci.yml b/.golangci.yml
index 4718eb4..5832a68 100644
--- a/.golangci.yml
+++ b/.golangci.yml
@@ -23,7 +23,9 @@ linters:
     - asciicheck
     - bidichk
     - bodyclose
+    - canonicalheader
     - contextcheck
+    - copyloopvar
     - decorder
     - dogsled
     - dupl
@@ -31,6 +33,7 @@ linters:
     - errcheck
     - errchkjson
     - errname
+    - fatcontext
     - ginkgolinter
     - gocheckcompilerdirectives
     - gochecksumtype
@@ -52,6 +55,7 @@ linters:
     - grouper
     - importas
     - ineffassign
+    - intrange
     - loggercheck
     - makezero
     - mirror
@@ -69,6 +73,7 @@ linters:
     - revive
     - rowserrcheck
     - sloglint
+    - spancheck
     - sqlclosecheck
     - staticcheck
     - stylecheck
@@ -86,6 +91,7 @@ linters:
     # - cyclop
     # - depguard
     # - dupword
+    # - err113
     # - errorlint
     # - exhaustive
     # - exhaustruct
@@ -98,13 +104,12 @@ linters:
     # - gochecknoinits
     # - gocognit
     # - godot
-    # - goerr113
-    # - gomnd
     # - inamedparam
     # - interfacebloat
     # - ireturn
     # - lll
     # - maintidx
+    # - mnd
     # - nestif
     # - nilerr
     # - nilnil
@@ -112,8 +117,8 @@ linters:
     # - noctx
     # - nonamedreturns
     # - paralleltest
-    # - testifylint
     # - tagliatelle
+    # - testifylint
     # - testpackage
     # - thelper
     # - tparallel
@@ -138,6 +143,7 @@ linters-settings:
       - badLock
       - badRegexp
       - badSorting
+      - badSyncOnceFunc
       - boolExprSimplify
       - builtinShadow
       - builtinShadowDecl
@@ -190,6 +196,7 @@ linters-settings:
       - stringsCompare
       - syncMapLoadAndDelete
       - timeExprSimplify
+      - todoCommentWithoutDetail
       - tooManyResultsChecker
       - truncateCmp
       - typeAssertChain
@@ -202,6 +209,7 @@ linters-settings:
       - unnecessaryDefer
       - weakCond
       - yodaStyleExpr
+      # - whyNoLint
   nolintlint:
     # Enable to ensure that nolint directives are all used. Default is true.
     allow-unused: false
```

Comment (cpanato) [reply dependencies.yaml:4]: done, thanks

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**